.github/workflows/appsignals-e2e-ec2-test.yml

```diff
@@ -65,16 +67,21 @@ jobs:
           aws-region: ${{ inputs.aws-region }}
 
       - name: Set up terraform
-        uses: hashicorp/setup-terraform@v3
-        with:
-          terraform_wrapper: false
+        run: |
+          source ${{ env.TEST_RESOURCES_FOLDER }}/.github/workflows/util/execute_and_retry.sh
+          execute_and_retry 2 "wget -O- https://apt.releases.hashicorp.com/gpg | sudo gpg --dearmor -o /usr/share/keyrings/hashicorp-archive-keyring.gpg"   
+          echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
+          sudo apt update && sudo apt install terraform
+
+      - name: Initiate Terraform
+        working-directory: terraform/ec2
+        run: |
+          source ${{ env.TEST_RESOURCES_FOLDER }}/.github/workflows/util/execute_and_retry.sh
+          execute_and_retry 2 "terraform init && terraform validate" "rm -rf .terraform && rm -rf .terraform.lock.hcl"
```

Comment:
Are we sure this is more reliable than the `hashicorp/setup-terraform` action? No point in making this change without testing it rigorously. I suggest running the test many times

Reply:
Regardless of whether it is more reliable or not, this error:
`(Set up Terraform) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit`
is a Node.js issue that we can't fix since `hashicorp/setup-terraform` uses Node.js to perform its action.

Reply:
Also, the commands used are the recommended installation step in Hashicorp's [website](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)

Additionally, we use wget to retrieve files in multiple instances during the e2e testing, so it will probably suffer from the same internet transient issue

Reply:
Got it, let's try it